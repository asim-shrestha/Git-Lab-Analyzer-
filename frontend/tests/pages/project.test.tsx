import React from 'react';
import {render} from '@testing-library/react';
import Index from '../../pages/server/[serverId]/projects';

describe("Project Folder", () =>{
    const useRouter = jest.spyOn(require('next/router'), 'useRouter');
    const mockUseEffect = jest.spyOn(React, 'useEffect')
    const mockAxios = jest.spyOn(require('axios'), 'get');

    it("Snapshot serverId", () => {
        useRouter.mockImplementationOnce(() => ({
            query: { serverId: 'TestId' },
        }));
        const { container } = render(
            <Index />
        )
        expect(container).toMatchSnapshot();

    })
    it("Test useEffect", ()=>{
        useRouter.mockImplementationOnce(() => ({
            query: { serverId: 'TestId' },
        }));
        render(<Index />);
        expect(mockUseEffect).toBeCalled();

    })

    it('Test axios',()=>{
        useRouter.mockImplementationOnce(() => ({
            query: { serverId: 'TestId' },
            isReady: true,
        }));
        render(<Index />);
        expect(mockAxios).toHaveBeenCalledTimes(1);
        useRouter.mockImplementationOnce(() => ({
            query: { serverId: 'TestId' },
            isReady: false,
        }));
        render(<Index />);
        expect(mockAxios).toHaveBeenCalledTimes(1);
    })


})