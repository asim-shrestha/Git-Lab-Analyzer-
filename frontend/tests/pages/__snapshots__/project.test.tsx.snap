// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Folder Snapshot serverId 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-backgroundGradient-1"
  >
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <div
        class="MuiBox-root MuiBox-root-5"
        id="card-header"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI1IiBoZWlnaHQ9IjEyNSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <img
            alt="The Gitlab Logo"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
        >
          GitLab
          <br />
          Analyzer
        </h6>
      </div>
      <div>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Loading projects...
        </p>
        <div
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
          role="progressbar"
        >
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
          />
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
